internal/build/phases.go

```diff
@@ -15,17 +15,32 @@ const (
 	platformDir    = "/platform"
 )
 
-func (l *Lifecycle) Detect(ctx context.Context, networkMode string, volumes []string) error {
-	detect, err := l.NewPhase(
+type RunnerCleaner interface {
```

Comment:
I get why this name was chosen but it's still slightly odd. Could it not simply be `Runner`?

Reply:
The go idiom of an interface called `Runner` would be to expect it to consist of a single method called `Run`. The compound equivalent is fairly common throughout the std library for interfaces of two or even three methods. It could possibly be shortened to something like `RunCleaner` which would be consistent with interface names like `ReadCloser`

Reply:
I did keep that in mind. The difference here being that it's not simply an interface for the composition of generic interfaces. It's an interface for a service type. There is no value to add `Cleaner` in my opinion given that it's a known and required function for the overall interface.

For example, [`Signal`](https://golang.org/pkg/os/#Signal) is not called `SignalStringer`. Another example, [`Listener`](https://golang.org/pkg/net/#Listener) is not called `ListenerCloser`.

I don't have a strong opinion, just a preference.

Reply:
I'm going to leave this as is for now, as the "cleanup" role of this interface seems essential to its function (given that we defer cleanup to ensure that it happens).